signer/src/hardware_service/api/mod.rs

```diff
@@ -71,12 +75,54 @@ pub async fn hardware_service_api(
 
 async fn hardware_service_api_inner(command: JsonCommandRequest) -> Result<JsonCommandResponse> {
     let response = match command {
-        JsonCommandRequest::get_account {} => {
+        JsonCommandRequest::get_account { fog_info } => {
             let account_info = hardware_service::get_account().await?;
             let hardware_account_id = hardware_service::get_account_id(account_info.clone());
+
+            let (default_public_address, change_public_address) = match fog_info {
+                Some(fog_info) => {
```

Comment:
I don't want to blow this PR up,  I've got a hunch we should probably move `AccountInfo` itself or something else to know if it's fog enabled and provide the fog addresses when requested. However guessing that's a significant overhaul to Full-Service

Reply:
Are you suggesting a struct similar to `AccountInfo` that bundles together the view private key + spend public key + public addresses?

Reply:
I'm probably not familiar enough with the code base yet to suggest anything meaningful
I think like how the PublicAddress has members for fog values and it knows how to look at or ignore those values when interacting with it.

I wouldn't worry about anything for now